pmd-core/src/test/java/net/sourceforge/pmd/cpd/XMLRendererTest.java

```diff
@@ -184,6 +188,37 @@ public void testWithOneDuplicationWithColumns() throws IOException {
         }
     }
 
+    @Test
+    public void testFilesWithNumberOfTokens() throws IOException {
+        final CPDReportRenderer renderer = new XMLRenderer();
+        final List<Match> matches = new ArrayList<>();
+        final String filename = "/var/Foo.java";
+        final int lineCount = 6;
+        final String codeFragment = "code\nfragment";
+        final Mark mark1 = createMark("public", filename, 1, lineCount, codeFragment, 2, 3);
+        final Mark mark2 = createMark("stuff", filename, 73, lineCount, codeFragment, 4, 5);
+        final Match match = new Match(75, mark1, mark2);
+        matches.add(match);
+        final Map<String, Integer> numberOfTokensPerFile = new HashMap<>();
+        numberOfTokensPerFile.put(filename, 888);
+        final CPDReport report = new CPDReport(matches.iterator(), numberOfTokensPerFile);
+        final StringWriter writer = new StringWriter();
+        renderer.render(report, writer);
+        final String xmlOutput = writer.toString();
+        try {
+            final Document doc = DocumentBuilderFactory.newInstance().newDocumentBuilder()
+                    .parse(new ByteArrayInputStream(xmlOutput.getBytes(ENCODING)));
+            final NodeList files = doc.getElementsByTagName("file");
+            final Node file = files.item(0);
+            final NamedNodeMap attributes = file.getAttributes();
+            assertEquals("/var/Foo.java", attributes.getNamedItem("path").getNodeValue());
+            assertEquals("888", attributes.getNamedItem("totalNumberOfTokens").getNodeValue());
+        } catch (Exception e) {
```

Comment:
You don't need this try-catch. If an exception occurs in the test, the test will fail automatically, including a nice stacktrace. Please remove.